docs/configuration/jobs.rst

```diff
@@ -173,6 +173,74 @@ Note that you will have to trigger replication manually using the ``zrepl signal
        type: manual
      ...
 
+.. _job-snapshotting-hooks:
+
+Pre- and post-snapshot Hooks
+----------------------------
+
+Jobs with `periodic snapshots <job-snapshotting-spec_>`_ can run an executable before and/or after taking the snapshot,
+configured in an optional ``hooks`` section of the ``snapshotting`` configuration:
+
+::
+
+
+    jobs:
+    - type: push
+      filesystems: {
+        "<": true,
+        "tmp": false
+      }
+      snapshotting:
+        type: periodic
+        prefix: zrepl_
+        interval: 10m
+        hooks:
+          pre: scripts/pre-snapshot.sh
+          post: /path/to/scripts/zrepl-notify.sh
+          timeout: 30s
+          keep_inconsistent: false
+      ...
+
+The path to the hook executables must either be absolute (e.g. ``/opt/bin/zrepl-notify.sh``) or is relative to the current configuration
+file directory (e.g. ``scripts/pre-snapshot.sh`` will be found in ``/etc/zrepl/scripts/pre-snapshot.sh``).
+No arguments may be specified; create a wrapper script if zrepl must call an executable that requires arguments.
+The configuration may specify a ``pre`` script only, a ``post`` script only, or both.
+
+The ``timeout`` parameter is optional and specifies a period after which zrepl will kill the hook process and report an error.
+The default is 30 seconds and may be specified in any units understood by `time.ParseDuration <https://golang.org/pkg/time/#ParseDuration>`_.
+
+The ``keep_inconsistent`` parameter specifies whether a snapshot will be taken after a failed pre-snapshot hook. The default
```

Comment:
`keep_` is a rather overloaded name already (keep rules for pruning)
Maybe something like this?:  `snap_on_err: true|false`
Or `snap: always|no_error`?
Or `hook_error: ignore | nosnap`.

I'm not sure about the default of skipping the snapshot:
If users monitor zrepl log output (either through log monitoring or the prometheus metrics), or if they monitor zrepl as a black box (i.e. 'are new snapshots being replicated, check using own script'), skipping snapshots is fine to me.
If they don't monitor at all, I guess an inconsistent DB snapshot might still be better than none at all, which is an argument against skipping by default.
Any ideas?